airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/metadata/pipeline.py

```diff
@@ -82,6 +82,7 @@ def __init__(
             title=title,
             context=context,
             paths_to_mount=[
+                MountPath(GIT_DIRECTORY_ROOT_PATH),
```

Comment:
Mounting `.git` is quite an expensive operation (depending on how many branches the local repo has).
As we have the git info in the context (`self.context.git_branch`, `self.context.git_revision`) what do you think about passing these to the step and remove the git logic from the metadata service lib?
It will have two advantages:
* remove the requirement for the metadata service lib to be executed on top of a git repo 
* keep a single source of truth about the current git info (stored in the context object)

Let me know what you think.

Reply:
I think I summed up my thoughts here:
https://github.com/airbytehq/airbyte/pull/37802#discussion_r1591602523

But I wanted to ask, what is the time penalty for mounting git in CI?

I ran locally with a fairly limited number of branches (since I reset my laptop) and it ran decently quick.

My thoughts were it wouldn't be the worst penalty since its during publish and not test

WDYT?

Reply:
The time penalty would rather happen locally, on CI we only checkout the current branch + master, which is not too expensive. 

> My thoughts were it wouldn't be the worst penalty since its during publish and not test

Agreed.